### PR TITLE
Implementation of set.equals

### PR DESCRIPTION
**Pull request by mvaligursky**
extracted from https://github.com/playcanvas/engine/pull/2990

### CODE REVIEW

**Comment by willeastcott** [comment]
I can't find this function in the JavaScript standard library:

https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Set

If `Set#isEqual` is not in the JS language spec, then this is not a polyfill.

**Comment by mvaligursky** [comment]
it's a polyfill from the future when JS implements this function ;)

do we have some better place to add these? Perhaps we could create a new folder for these. Or just add it to "/core"?

**Comment by willeastcott** [comment]
My feeling is that if we are not polyfilling a legitimately existing JS language feature/function, we shouldn't screw with standard library classes, poking in our own stuff. You _could_ stick a utility function in `core\set-utils.js` or something? But yeah, operate _on_ set instances rather than inject stuff into `Set`.

File: src/index.js [hunk 1]
```diff
@@ -17,6 +17,7 @@ export { events } from './core/events.js';
 export { guid } from './core/guid.js';
 export { path } from './core/path.js';
 export { platform } from './core/platform.js';
+export { set } from './core/set-utils.js';
```
Comment (willeastcott) [inline src/index.js:20]: This shouldn't be exported to the pc namespace - it's internal.

File: tests/core/test_set_utils.js [hunk 2]
```diff
@@ -0,0 +1,13 @@
...
+        var set2 = new Set([2, 1, 3]);
+        var set3 = new Set([2, 1, 6]);
+        var set4 = new Set([1, 2]);
+
+        expect(pc.set.isEqual(set1, set2)).to.equal(true);
+        expect(pc.set.isEqual(set1, set3)).to.equal(false);
+        expect(pc.set.isEqual(set1, set4)).to.equal(false);
+    });
+
+});
```
Comment (willeastcott) [inline tests/core/test_set_utils.js:13]: If this is internal, I don't think you can unit test it. Although you can unit test any public function that uses it of course. Might have to delete this.

**Comment by Maksims** [comment]
Feels like it could be better named, consistent with Vec3 for example: `equals`.

**Review by willeastcott — APPROVED** [review]

**Pull request merged**